### PR TITLE
Merge the vSphere builder

### LINKED ISSUE #4526: investigate using vSphere API
There appear to be some limitations with how we work with vSphere, where the solution seems to be using the API directly.

Will use this as a meta-ticket to track the feasibility of that.

* https://github.com/mitchellh/packer/issues/2715
* #2841

### LINKED ISSUE #8532: Packer 1.5.0 and 1.5.1 cannot start vpshere-iso
#### Overview of the Issue

Newest versions of Packer - 1.5.0 and 1.5.1 don't work with vsphere-iso builder

#### Reproduction Steps

1. Trying to build Ubuntu 18.04 template with Packer 1.5.0 and 1.5.1 using vsphere-iso 2.3 - error
2. Installed earlier version 1.4.5 - issues with uploading the ISO
3. Installed earlier version 1.3.5 - managed to build the template successfully. However, preseeding file did not load properly, so I had to manually configure the Ubuntu OS.

### Packer version
1.5.0
1.5.1

### Simplified Packer Build-file

"builders": [
    {
      "type": "vsphere-iso",

### Operating system and Environment details

Windows 10

### Log Fragments and crash.log files

Failed to initialize build 'vsphere-iso': error initializing builder 'vsphere-iso': exec: "C:\\Packer": file does not exist

==> Builds finished but no artifacts were created.

### PR DESCRIPTION
**Pull request by randomcamel**
This is to merge [the vSphere builder](https://github.com/jetbrains-infra/packer-builder-vsphere) written 
by JetBrains and others.

Closes #8532
Closes #4526

### CODE REVIEW

**Comment by SwampDragons** [comment]
Thanks for opening this!

**Comment by azr** [comment]
2 signatures to go 🎉

**Comment by azr** [comment]
@remijouannet, @scbjans; sorry for pinging you, we only need you two to [sign the cla](https://cla.hashicorp.com/hashicorp/packer?pullRequest=8480) so that your work can be incorporated into Packer 🙂 then, we can/will help maintain it onwards !!! 😄.

How does that sound to you ?

**Comment by remijouannet** [comment]
hello, CLA sign

**Comment by SwampDragons** [comment]
Thanks so much for signing the CLA, folks! I'll take it from here.

**Comment by marema31** [comment]
@SwampDragons , If you need for your tests , I can give you access to a vSphere.

**Comment by SwampDragons** [comment]
Thanks, I'm working on getting some set up internally, just takes a bit of time :D

**Comment by SwampDragons** [comment]
Plus, normally we don't run acceptance tests on every PR, just on releases. So I'll be reworking the organization a bit.

File: builder/vsphere/iso/builder.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+	"github.com/hashicorp/packer/builder/vsphere/common"
+	"github.com/hashicorp/packer/builder/vsphere/driver"
+	packerCommon "github.com/hashicorp/packer/common"
+	"github.com/hashicorp/packer/helper/communicator"
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+)
+
+type Builder struct {
+	config *Config
```
Comment (azr) [inline builder/vsphere/iso/builder.go]:
```suggestion
	config Config
```

**Comment by azr** [comment]
Yaaaaay 👍 👍 👍 👍 👍

**Pull request merged**